src/scripts/create_jentoft.py

```diff
@@ -0,0 +1,180 @@
+
+    # Remove the dataset from Hugging Face Hub if it already exists
+    try:
+        api: HfApi = HfApi()
+        api.delete_repo(dataset_id, repo_type="dataset")
+    except HTTPError:
+        pass
+
+    # Push the dataset to the Hugging Face Hub
+    dataset.push_to_hub(dataset_id, private=True)
+
+
+def prepare_dataset(dataset_url: str) -> pd.DataFrame:
+    """Prepare the dataset from a URL.
+
+    Args:
+        dataset_url:
+            The URL to the dataset.
+
+    Returns:
+        The prepared dataset.
+    """
+    df = pd.read_csv(dataset_url, sep="\t", on_bad_lines="skip")
+
+    # Make label column
+    df["label"] = df["ERROR"].apply(
+        lambda x: "correct" if x == "no_error" else "incorrect"
+    )
+
+    # Rename the columns
+    df = df.rename(columns={"SOURCE": "text"})
+
+    # Only keep relevant columns
+    df = df[["text", "label"]]
+
+    # Remove text duplicates
+    df = df.drop_duplicates(subset=["text"])
+
+    # Remove samples with with 5 or fewer words
+    df = df[df["text"].str.split().apply(len) > 5]
+    return df
+
+
+def sample_dataset(full_df: pd.DataFrame, size: int) -> pd.DataFrame:
+    """Sample a dataset with an equal number of 'correct' and 'incorrect' samples.
+
+    Args:
+        full_df:
+            The full dataset.
+        size:
+            The size of the sampled dataset.
+
+    Returns:
+        The sampled dataset.
+    """
+    half_size = size // 2
+    assert (
+        full_df["label"] == "correct"
+    ).sum() >= half_size, "Not enough 'correct' samples"
+
+    correct_df = full_df[full_df["label"] == "correct"].sample(
+        n=half_size, random_state=4242
+    )
+    incorrect_df = full_df[full_df["label"] == "incorrect"].sample(
+        n=half_size, random_state=4242
+    )
+    return pd.concat([correct_df, incorrect_df])
+
+
+def remove_overlapping_samples(
+    full_train_df: pd.DataFrame, full_val_df: pd.DataFrame, full_test_df: pd.DataFrame
+) -> tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    """Ensure that a sample is only present in one split.
+
+    Args:
+        full_train_df:
+            The full training dataset.
+        full_val_df:
+            The full validation dataset.
+        full_test_df:
+            The full test dataset.
+
+    Returns:
+        The cleaned training, validation, and test datasets.
+    """
+    train_texts = set(full_train_df["text"])
+    val_texts = set(full_val_df["text"])
+    test_texts = set(full_test_df["text"])
+
+    # Find overlaps
+    train_val_overlap = train_texts.intersection(val_texts)
+    train_test_overlap = train_texts.intersection(test_texts)
+    val_test_overlap = val_texts.intersection(test_texts)
+
+    # Print overlap statistics
+    print(f"Overlapping texts between train and validation: {len(train_val_overlap)}")
+    print(f"Overlapping texts between train and test: {len(train_test_overlap)}")
+    print(f"Overlapping texts between validation and test: {len(val_test_overlap)}")
```

Comment:
Can you instead create a logger, and use `logger.info` here instead of printing?